src/services/shims.ts

```diff
@@ -270,6 +271,10 @@ module ts {
             return this.shimHost.getProjectVersion();
         }
 
+        public useCaseSensitiveFileNames(): boolean {
+            return this.shimHost.useCaseSensitiveFileNames && this.useCaseSensitiveFileNames();

```

Comment:
i would test this to make sure it works with the managed side.. I remember there were some IDisbatch issues with looking up non-existing methods.